pelican/writers.py

```diff
@@ -55,7 +55,9 @@ def _add_item_to_the_feed(self, feed, item):
             categories=item.tags if hasattr(item, 'tags') else None,
             author_name=getattr(item, 'author', ''),
             pubdate=set_date_tzinfo(
-                item.modified if hasattr(item, 'modified') else item.date,
+                item.date, self.settings.get('TIMEZONE', None)),
+            pubdate=set_date_tzinfo(
+                item.modified if hasattr(item, 'modified') else None,
                 self.settings.get('TIMEZONE', None)))
```

Comment:
`set_date_tzinfo(None, [...])` may throw exceptions or does not return `None`.

How about:
```
=set_date_tzinfo(item.modified, self.settings.get('TIMEZONE', None)) if hasattr(item, 'modified') else None
```